SimpleInstapaperKit/IKLoginViewController.m

```diff
@@ -115,23 +129,31 @@ - (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
 }
 
 - (void)setupColors {
-    self.backgroundColor = [UIColor colorWithRed:roundf(243.f / 255.f) green:roundf(245.f / 255.f) blue:roundf(247.f / 255.f) alpha:1.f];
+    self.backgroundColor = [UIColor colorWithRed:255 green:255 blue:255 alpha:1.f];
     
-    self.cellBackgroundColor = [UIColor colorWithRed:roundf(243.f / 255.f) green:roundf(245.f / 255.f) blue:roundf(247.f / 255.f) alpha:1.f];
-    self.cellBackgroundColorSelected = [UIColor colorWithRed:roundf(214.f / 255.f) green:roundf(219.f / 255.f) blue:roundf(223.f / 255.f) alpha:1.f];
+    self.cellBackgroundColor = [UIColor colorWithRed:255 green:255 blue:255 alpha:1.f];
+    self.cellBackgroundColorSelected = [UIColor colorWithRed:242 green:242 blue:247 alpha:1.f];
     
-    self.primaryTextColor = [UIColor colorWithRed:roundf(0.f / 255.f) green:roundf(0.f / 255.f) blue:roundf(0.f / 255.f) alpha:1.f];
-    self.secondaryTextColor = [UIColor colorWithRed:roundf(141.f / 255.f) green:roundf(149.f / 255.f) blue:roundf(158.f / 255.f) alpha:1.f];
-    self.linkTextColor = [UIColor colorWithRed:roundf(0.f / 255.f) green:roundf(122.f / 255.f) blue:roundf(255.f / 255.f) alpha:1.f];
+    self.primaryTextColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:1.f];
+    self.secondaryTextColor = UIColor.darkGrayColor;
+    self.placeholderTextColor = UIColor.lightGrayColor;
+    self.linkTextColor = [UIColor colorWithRed:0 green:122 blue:255 alpha:1.f];
+    
+    self.clearButtonTintColor = UIColor.lightGrayColor;
 }
 
 - (void)viewDidLoad
 {
     [super viewDidLoad];
-
     [self.view addSubview:self.loadingField];
     
     self.tableView.backgroundColor = self.backgroundColor;
+    
+    self.usernameFieldClearButton.tag = kUsernameFieldClearButtonTag;
+    self.passwordFieldClearButton.tag = kPasswordFieldClearButtonTag;
+    
+    [self.usernameField addTarget:self action:@selector(textFieldTextDidChange:) forControlEvents:UIControlEventEditingChanged];
+    [self.passwordField addTarget:self action:@selector(textFieldTextDidChange:) forControlEvents:UIControlEventEditingChanged];
```

Comment:
This project don't use ARC.. so I wonder, do we need to remove target in some place?